pkg/tagger/collectors/garden_main.go

```diff
@@ -0,0 +1,108 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-Present Datadog, Inc.
+
+package collectors
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/cloudfoundry"
+	"github.com/DataDog/datadog-agent/pkg/util/clusteragent"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/retry"
+)
+
+const (
+	cfCollectorName = "cloudfoundry"
+)
+
+// GardenCollector collects tags for CF application containers
+type GardenCollector struct {
+	infoOut             chan<- []*TagInfo
+	dcaClient           clusteragent.DCAClientInterface
+	gardenUtil          cloudfoundry.GardenUtilInterface
+	clusterAgentEnabled bool
+}
+
+// Detect tries to connect to the Garden API and the cluster agent
+func (c *GardenCollector) Detect(out chan<- []*TagInfo) (CollectionMode, error) {
+
+	// Detect if we're on a compute VM by trying to connect to the local garden API
+	var err error
+	c.gardenUtil, err = cloudfoundry.GetGardenUtil()
+	if err != nil {
+		if retry.IsErrWillRetry(err) {
```

Comment:
Do you need this condition when you `return NoCollection, err` anyway?

Reply:
nope I don't